app/views/comments/_edit.html.erb

```diff
@@ -1,153 +1,153 @@
-        e.preventDefault();
-        $('#c<%= comment.id%>div').addClass('hover');
-      });
-
-      $('#c<%= comment.id%>div').bind('dragout',function(e) {
-        $('#c<%= comment.id%>div').removeClass('hover');
-      });
-
-      $('#c<%= comment.id%>div').bind('drop',function(e) {
-        e.preventDefault();
-        setInit(<%= comment.id %>);
-      });
-
-      $('#c<%= comment.id%>div').fileupload({
-        url: "/images",
-        paramName: "image[photo]",
-        dropZone: $('#c<%= comment.id%>div'),
-        dataType: 'json',
-        formData: {
-          'uid':<%= current_user.uid %>,
-          'nid':<%= comment.uid %>
-        },
-        start: function(e) {
-          $('#c<%= comment.id%>progress').show()
-          $('#c<%= comment.id%>uploading').show()
-          $('#c<%= comment.id%>prompt').hide()
+      <script type="application/javascript">
+        jQuery(document).ready(function() {
+          $E.initialize();
+        });
+
+        function setInit(id) {
+          var args = {
+            'textarea': 'c'+id+'text',
+            'preview': 'c'+id+'preview'
+          }
+          $E.initialize(args);
+        }
+
+        $('#c<%= comment.id%>div').bind('dragover',function(e) {
+          e.preventDefault();
+          $('#c<%= comment.id%>div').addClass('hover');
+        });
+
+        $('#c<%= comment.id%>div').bind('dragout',function(e) {
           $('#c<%= comment.id%>div').removeClass('hover');
-        },
-        done: function (e, data) {
-          $('#c<%= comment.id%>progress').hide()
-          $('#c<%= comment.id%>uploading').hide()
-          $('#c<%= comment.id%>prompt').show()
-          var is_image = false
-          if (data.result['filename'].substr(-3,3) == "jpg") is_image = true
-          if (data.result['filename'].substr(-4,4) == "jpeg") is_image = true
-          if (data.result['filename'].substr(-3,3) == "png") is_image = true
-          if (data.result['filename'].substr(-3,3) == "gif") is_image = true
-          if (data.result['filename'].substr(-3,3) == "JPG") is_image = true
-          if (data.result['filename'].substr(-4,4) == "JPEG") is_image = true
-          if (data.result['filename'].substr(-3,3) == "PNG") is_image = true
-          if (data.result['filename'].substr(-3,3) == "GIF") is_image = true
-
-          if (is_image) {
-            image_url = data.result.url.split('?')[0];
-            orig_image_url = image_url.replace('medium','original');
-            $E.wrap('[![',']('+image_url+')]('+orig_image_url+')', {'newline': true, 'fallback': data.result['filename']});
-          } else {
-            $E.wrap('<a href="'+data.result.url.split('?')[0]+'"><i class="fa fa-file"></i> ','</a>', {'newline': true, 'fallback': data.result['filename']});
+        });
+
+        $('#c<%= comment.id%>div').bind('drop',function(e) {
+          e.preventDefault();
+          $D.selected = $(e.target).closest('div.comment-form-wrapper').eq(0);
```

Comment:
Notice this additional line!

Reply:
I wrapped everything in _edit.html.erb in a div.comment-form-wrapper.

The other major change is that now $D.selected is assigned in this eventListener bound to #c1div.

A major part of the drag-and-drop crosswiring is fixed. My tests (including the new one I wrote for this issue) are passing. I had to rewrite some of the system tests in this most recent push: 'edit comment' and 'delete comment' rely on .comment-form-wrapper for some reason.

Reply:
Oops, the tests aren't passing. I'll take a closer look tomorrow.

Reply:
this is awesome